iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/FunctionDeclarationImplTest.java

```diff
@@ -32,23 +33,26 @@
 class FunctionDeclarationImplTest extends BicepTreeModelTest {
 
   @Test
-  void shouldParseFunctionDelaration() {
+  void shouldParseFunctionDeclaration() {
     ArmAssertions.assertThat(BicepLexicalGrammar.FUNCTION_DECLARATION)
-      .matches("func myFunction lambdaExpression")
-      .matches("func myFunction   lambdaExpression")
+      .matches("func myFunction() string => 'result'")
+      .matches("func myFunction () string =>   'result'")
+      .matches("func myFunction(foo int) string => '${foo}'")
+      .matches("func myFunction(foo int, bar object) int => 0")
 
+      .notMatches("func myFunction() => 'result'")
       .notMatches("func myFunction")
       .notMatches("func myFunction = lambdaExpression")
       .notMatches("func");
   }
 
   @Test
   void shouldParseSimpleFunctionDeclaration() {
-    String code = code("func myFunction lambdaExpression");
+    String code = code("func myFunction() string => 'result'");
```

Comment:
Maybe add another assert:
`assertThat(tree.lambdaExpression().is(ArmTree.Kind.TYPED_LAMBA_EXPRESSION)).isTrue();`